microservices-orders/scripts/run-services.sh

```diff
@@ -12,6 +12,14 @@ ADDITIONAL_ARGS=${ADDITIONAL_ARGS:-""}
 LOG_DIR=${LOG_DIR:="logs"}
 PIDS_FILE=${PIDS_FILE:=".microservices.pids"}
 
+# check if we can write to the log directory
+if touch $LOG_DIR/permissions_test; then
+  rm $LOG_DIR/permissions_test
+else
+  echo "Unable to write to the logs dir with uid `id -u`. Logging to /tmp."
+  LOG_DIR=/tmp
+fi
```

Comment:
@russau what do you think about something like this instead of writing and deleting the temporary file?

```suggestion
if [ ! -w $LOG_DIR ]; then
  echo "Unable to write to the logs dir with uid `id -u`. Logging to /tmp."
  LOG_DIR=/tmp
fi
```

Reply:
I tried the `-w` test when I first wrote this and for some reason it was saying I *do* have write access!  Then a stackoverflow article persuaded me over to the temp file.  I have a linux box up that I can do another test on.  This would be much neater - let me double check why I gave up on this.

Reply:
I've tested the change on mac and linux - all works, much neater!  I've amended the commit and pushed.